src/lib/pk_pad/eme_pkcs1/eme_pkcs.cpp

```diff
@@ -27,17 +28,16 @@ secure_vector<uint8_t> EME_PKCS1v15::pad(const uint8_t in[],
    }
 
    secure_vector<uint8_t> out(key_length);
+   BufferStuffer stuffer(out);
 
-   out[0] = 0x02;
-   rng.randomize(out.data() + 1, (key_length - inlen - 2));
+   const size_t padding_bytes = key_length - inlen - 2;
 
-   for(size_t j = 1; j != key_length - inlen - 1; ++j) {
-      if(out[j] == 0) {
-         out[j] = rng.next_nonzero_byte();
-      }
+   stuffer.append(0x02);
+   for(size_t i = 0; i != padding_bytes; ++i) {
+      stuffer.append(rng.next_nonzero_byte());
    }
```

Comment:
The existing approach was really a stupid microoptimization, on the basis that we would be statistically somewhat likely to not need to invoke the RNG multiple times, saving a few virtual calls. I like the new version because it more cleanly matches to what a standard algorithmic description of the padding looks like, and the performance is totally irrelevant considering the cost of even an RSA encryption swamps any optimizations we could make here.

Reply:
Full ack.